### PR TITLE
[WIP] idist - Docs & code updates

### PR DESCRIPTION
**Pull request by vfdev-5**
Description:

- docs updates
- updated metrics and their tests
- get local_rank with hostname approach
- added `idist.all_gather`

Check list:
* [ ] New tests are added (if a new feature is added)
* [ ] New doc strings: description and/or example code are in RST format
* [ ] Documentation is updated (if required)

### CODE REVIEW

File: ignite/distributed/utils.py [hunk 1]
```diff
@@ -302,11 +324,12 @@ def initialize(backend: str, **kwargs):
 
     Examples:
 
-        Launch single node multi-GPU training with `torch.distributed.launch` utility:
+        Launch single node multi-GPU training with `torch.distributed.launch` utility. Please, note that we need to
```
Comment (sdesrozis) [inline ignite/distributed/utils.py:327]: No need to specify `--use_env`

**Comment by vfdev-5** [comment]
@sdesrozis I think we are good for this PR into `idist`. Please review

File: ignite/metrics/metric.py [hunk 2]
```diff
@@ -46,9 +40,6 @@ def __init__(self, output_transform: Callable = lambda x: x, device: Optional[Un
                     "across all computing devices".format(self.__class__.__name__),
                     RuntimeWarning,
                 )
-            if device is None:
-                device = "cuda"
-            device = torch.device(device)
         self._device = device
```
Comment (sdesrozis) [inline ignite/metrics/metric.py:43]: Is `_device` used ?

Comment (vfdev-5) [reply ignite/metrics/metric.py:43]: Yes, it is used in certain metrics that proposes to store internals on a device.

**Pull request merged**